airbyte-ci/connectors/pipelines/tests/conftest.py

```diff
@@ -23,8 +23,12 @@ def anyio_backend():
 
 
 @pytest.fixture(scope="module")
-async def dagger_client():
-    async with dagger.Connection(dagger.Config(log_output=sys.stderr)) as client:
+def dagger_connection():
+    return dagger.Connection(dagger.Config(log_output=sys.stderr))
```

Comment:
❓ I think this makes sense and is fine, but don't see it used anywhere, just wondering why we split this

Reply:
Ahh I can merge back together. There was an approach I was using to mocking out our `get_dagger_client` method in the `ClickPipelineCOntext` which needed the dagger connection as a fixture instead of the client.

I didnt end up going that route for the test in the end